### PR TITLE
Provides full testing

### PR DESCRIPTION
**Pull request by davidgaspardev**
Following the exercise description to provide a complete test.

### CODE REVIEW

File: exercises/practice/two-fer/src/test/kotlin/TwoFerTest.kt [hunk 1]
```diff
@@ -10,21 +10,18 @@ class TwoFerTest {
     }
 
     @Test
-    @Ignore
```
Comment (SleeplessByte) [inline exercises/practice/two-fer/src/test/kotlin/TwoFerTest.kt:13]: This should be ignored as in practice exericses, all tests are ignored except for the first.

File: exercises/practice/two-fer/src/test/kotlin/TwoFerTest.kt [hunk 2]
```diff
@@ -10,21 +10,18 @@ class TwoFerTest {
...
     fun aNameGiven() {
         assertEquals("One for Alice, one for me.", twofer("Alice"))
     }
 
     @Test
-    @Ignore
```
Comment (SleeplessByte) [inline exercises/practice/two-fer/src/test/kotlin/TwoFerTest.kt:19]: This should be ignored as in practice exericses, all tests are ignored except for the first.

File: exercises/practice/two-fer/src/test/kotlin/TwoFerTest.kt [hunk 3]
```diff
@@ -10,21 +10,18 @@ class TwoFerTest {
...
     fun anotherNameGiven() {
         assertEquals("One for Bob, one for me.", twofer("Bob"))
     }
 
     @Test
-    @Ignore
```
Comment (SleeplessByte) [inline exercises/practice/two-fer/src/test/kotlin/TwoFerTest.kt:25]: This should be ignored as in practice exericses, all tests are ignored except for the first.

File: exercises/practice/two-fer/src/test/kotlin/TwoFerTest.kt [hunk 4]
```diff
@@ -10,21 +10,18 @@ class TwoFerTest {
...
     fun emptyStringGiven() {
-        assertEquals("One for , one for me.", twofer(""))
+        assertEquals("One for you, one for me.", twofer(""))
```
Comment (SleeplessByte) [inline exercises/practice/two-fer/src/test/kotlin/TwoFerTest.kt:24]: Good catch, thanks!

**Review by SleeplessByte — CHANGES_REQUESTED** [review]

**Comment by kotp** [comment]
On 21/12/26 07:18PM, Derk-Jan Karrenbeld wrote:
> @SleeplessByte requested changes on this pull request.
> >      fun emptyStringGiven() {
> -        assertEquals("One for , one for me.", twofer(""))
> +        assertEquals("One for you, one for me.", twofer(""))
> 
> Good catch, thanks!

If a name is given it should be used.  If an empty string is given as
the name, then it is implied by "If not given a name, use 'you'" part of
the description (unless this has changed.)  If `nil` or `null` is given,
then that should be used, if `42` is given, then that should be used.

From: https://github.com/exercism/kotlin/blob/main/exercises/practice/two-fer/.docs/instructions.md

> Where "name" is the given name.
>
> However, if the name is missing, return the string:

The only time that '"you"' should be used, is when no name is given, at
least that is how I interpret this, and as far as I can see the existing
tests do not contradict this, though it also does not test either way,
which I find is a benefit for mentoring, rather than a weakness.

**Comment by SleeplessByte** [comment]
Yeah, i don't mind keeping it for that reason. It's whatever you all want for mentoring 🙃😊.

If we were to follow Kotlin practices and make the argument optional, '' would not be the same as no string.

File: exercises/practice/two-fer/src/test/kotlin/TwoFerTest.kt (hunk 4 above)
Comment (kotp) [reply exercises/practice/two-fer/src/test/kotlin/TwoFerTest.kt:24]: Let's keep this as it is, without the 'you' if truly we are giving an
empty string as a name.  (The giver of the name is the name expert,
after all, and we asked for it, after all.  Perhaps their parents named
them `""` or they legally changed their name to `""` (good luck with
that, but then Prince…))

**Comment by SleeplessByte** [comment]
In that case, none of the changes are applicable.

**Comment by kotp** [comment]
Thank you @davidgaspardev for bringing the proposed changes, as it helps to get some of the discussions out there as to why the code is the way it is.

**Pull request closed without merging**